jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/ProgressEventDispatcherWrapper.java

```diff
@@ -65,8 +57,9 @@ public void close() {
     progressEventDispatcher.close();
   }
 
-  void initializeWithBlobSize(long blobSize) {
+  void setProgressTarget(long allocationUnits) {
```

Comment:
Seems a bit weird to have to call the constructor and this method to fully initialize a `ProgressEventDispatcherWrapper`, maybe we can consolidate this by changing the signature of `RegistryClient#pullBlob`?

Reply:
Yeah, I did not like this wrapper either. The problem Q originally faced was the progress reporting design that

1. You have to know and set in advance how much progress units you will allocate when creating a progress event dispatcher.
1. But you cannot know how much bytes (allocation units) you will download until you read `Content-Length`.
1. So the wrapper that will create and hold a progress event dispatcher when this method is called. Other than that, this class is just a thin wrapper that delegates `dispatchProgress()`.

For this reason, obviously it's best to get rid of this unfortunate wrapper if possible. As an attempt to work around the issue, Q probably also felt it doesn't make much sense to have this wrapper in the signature and make it a hard dependency from the API standpoint, also given that there are only just a couple places this wrapper is required. And as I said in the earlier comment, my intention is to actually decouple progress dispatcher stuff from `RegistryClient`.